DOCS-6157-added-links (#8)
mulesoft/docs-service-mesh: modules/ROOT/pages/download-and-install-service-mesh.adoc

image::install-service-mesh-breadcrumb.png[]

The following topics are covered in this section:

* <<download-service-mesh,Download Service Mesh Client>>
* <<install-service-mesh,Install Service Mesh>>

> 

Reviewer: GLOBALLY replace this text in the beginning of each file, for the reasons mentioned earlier.

> Before you install Service Mesh, you must first download the product from the installer download URL. As part of installation, the following required components are deployed on a Kubernetes cluster:

Reviewer: ```suggestion
Before you install Anypoint Service Mesh, download it from the installer download URL and deploy the following required components on a Kubernetes cluster:
```

Reviewer: Actually, now that I think further about this, THIS topic set includes downloading the client and installing (and uninstalling) the product. All other tasks are either outside of these or subordinate to them, so we should not be including the Kubernetes cluster components downloading and deployment, unless we have a task or requirements topic devoted to this.

The intro text here should focus on the title: download and install, and why/when/under what circumstances the audience wants or needs to perform these tasks. The subtopics are download, install, uninstall. Any extra configuration or deployment needs its own topic set or subtopic.

* Istio
* Service Catalog

> * Mulesoft Anypoint Service Mesh Broker

Reviewer: I doubt that we will be including MuleSoft in these names.... Our product is Anypoint.

> * Mulesoft ClusterIP Service

Reviewer: This sounds like a generic item, not a product; in fact, we never capitalize service names.

 

[[download-service-mesh]]

> == Download Anypoint Service Mesh Client

Reviewer: HIDDEN TEXT EDIT

You install, manage, and troubleshoot your Anypoint Service Mesh installation by using the Service Mesh client. To download the client, type the following at the command prompt:

`curl -Ls http://anypoint.mulesoft.com/servicemesh/xapi/v1/install > service-mesh && chmod +x service-mesh`

The Service Mesh client is downloaded on your local drive.

[[install-service-mesh]]
== Install Anypoint Service Mesh

When you invoke the installer, Istio is installed first, and then the Mule adapter and broker are installed inside the Istio mixer. To install Anypoint Service Mesh, perform the following steps:

. From the location where you downloaded your Service Mesh client, type `./service-mesh install`.

. Enter your client ID.

. Enter your client secret.

. Enter your MuleSoft license number.
+
Anypoint Service Mesh installs within few minutes and displays a status:
+
image::service-mesh-installer-output.png[Example Installation Output]
+
If an installation error occurs, the installer times out and uninstalls all previously installed Service Mesh components. 
. Verify the status of all pods:
+
`> kubectl get pods -n service-mesh`
+
MuleSoft Anypoint Service Mesh is installed and the status of all pods is displayed:
+
image::verified-service-mesh-installer-output.png[]

== Uninstall Service Mesh

You might have to uninstall or delete Anypoint Service Mesh: for example, to troubleshoot issues. 

Before you uninstall Service Mesh:

. Delete all bindings.
. Delete all adapters.
. Remove the injection label on the namespaces with provisioned applications.
. Redeploy your applications.

To uninstall Service Mesh, run the following command from the installer download URI:

`> ./service-mesh uninstall && rm service-mesh`
